### DISCUSSION TITLE
InfluxDB v2 integration help

### DISCUSSION THREAD

**Discussion by ddyer8** [tagyoureit/nodejs-poolController#506] [Q&A]
Does influxDB v2 work with Nixie? 

Are there setup instructions for piping the pool data to influx like other items in the wiki?

I have influxDB installed, but I could use some pointers to connecting to Nixie (beyond just enabling the connection)

**Comment by tagyoureit** [comment]
https://github.com/tagyoureit/nodejs-poolController/wiki/Bindings-Integrations-in-2.0#influx

Let us know if you have specific questions.

**Reply by tagyoureit** [reply]
I just accidentally blew away my Influx v1.8 container and it was updated auto-magically to v2.3.  I decided to see if anything has changed since my last experience and here are some thoughts...

1. It isn't easy to connect Grafana to Influx v2.x.  After scratching my head for a while, I remembered I found this post previously and it really helped:  https://community.grafana.com/t/trouble-adding-influxdb-2-0-as-datasource/36490/29
2. Influx is moving to Flux.  That's all fine and good, but Grafana is pretty far behind.  The [Getting Started Guide](https://grafana.com/docs/grafana/next/getting-started/get-started-grafana-influxdb/) literally has Step 5 as "cut the code from Influx and paste it into Grafana".  `In Grafana, add a panel and then paste your Flux code into the query editor.` 
3. I can't seem to connect to Grafana using InfluxQL (old language) at all.  It throws errors and I haven't figured out how to get past this yet.

All of this means I'll need to start over.  I'll try to recreate some of my older dashboards with Flux. 

Let me know how you are coming along.

**Reply by johnny2678** [reply]
This. This right here is why you will have to pry influx 1.8.4 from my cold dead hands.  Might as well move to Prometheus or some other TS database because it's like starting over.

 and I've got wayyy to much time in grafana dashboard development written for Influx 1.8.x

Would be happy for someone to show me the way to 2.x that isn't a massive timesuck but i just don't see it and Influx 1.8.x works fine.

**Comment by ddyer8** [comment]
I won’t have time to work on it for a couple days at least.  Should I do something different than influx v2.3?

**Reply by tagyoureit** [reply]
Well... it just took me another hour to figure out how to do a simple math calculation to get the remaining acid tank capacity.

In Influx v1.8, this is the syntax:
```
SELECT mean("tank level") / mean("tank capacity")  FROM "chemControllersPh" 
WHERE $timeFilter GROUP BY time($__interval) fill(null)
```

In Influx v2.0+ (and I got the correct answer, but unsure if this is the best way to do it):
```
from(bucket: "pool")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "chemControllersPh")
  |> filter(fn: (r) => r["_field"] == "tank level" or r["_field"] == "tank capacity")
  |> aggregateWindow(every: v.windowPeriod, fn: mean, createEmpty: false)
  |> pivot(rowKey:["_time"], columnKey: ["_field"], valueColumn: "_value")
  |> map( fn: (r) => ({_value: r["tank level"] / r["tank capacity"]}))
  |> last()
```

If you like jumping in and solving new problems and blazing new paths, you can go with Influx v2.3 and we can try to figure it out together.  If you want a simpler path with more existing knowledge, then Influx v1.8 is for you.

**Comment by ddyer8** [comment]
I have influx v2 setup and running, but I haven't done much with it.  I may be in a better spot since I'm not trying to migrate legacy dashpanels and don't know any of it lol.

I am getting a persistent error influx logs about an issue binding heatMode.  I don't have heat on the pool, so should I just remove some of this from the logging config?

3|njsPC      | [8/3/2022, 19:29:05] error: InfluxDB tag binding failure on body:heatMode/@bind=data.heatMode.desc; --> @bind=data.heatMode.desc;  {"id":1,"heaterOptions":{"total":0,"gas":0,"solar":0,"heatpump":0,"ultratemp":0,"hybrid":0,"mastertemp":0,"maxetherm":0},"isCovered":false,"heaterCooldownDelay":false,"startDelay":false,"stopDelay":false,"showInDashboard":true,"name":"Pool","setPoint":0,"circuit":6,"type":{"val":0,"name":"pool","desc":"Pool"},"heatStatus":{"val":0,"name":"off","desc":"Off"},"isOn":true,"temp":88.73}

**Comment by rstrouse** [comment]
>  I don't have heat on the pool, so should I just remove some of this from the logging config?

You can.  I would suggest copying the existing file and performing edits to a custom interface.  This will make sure the changes you make do not get overwritten by software updates to the influxDB.json and git will not complain about it.
